providers/file.py

```diff
@@ -1,14 +1,40 @@
 import logging
 from domain import Domain
+from os import listdir
+from os.path import isfile, join
 
-description = "Read domains from a file, one per line"
+description = "Read domains from a file (or folder of files), one per line"
 
 
 def fetch_domains(filename, **args):
-    with open(filename) as file:
-        try:
-            lines = file.readlines()
-            logging.warn(f"Ingested {len(lines)} domains from file '{filename}'")
-        except Exception as e:
-            logging.error(f"Could not read any domains from file {filename} -- {e}")
-    return [Domain(line.rstrip()) for line in lines]
+    if isfile(filename):
+        with open(filename) as file:
+            try:
+                domains = file.readlines()
+                logging.warn(f"Ingested {len(domains)} domains from file '{filename}'")
+            except Exception as e:
+                logging.error(f"Could not read any domains from file {filename} -- {e}")
+                exit(-1)
+    else:
+        domains = []
+        files = fetch_nested_files(filename)
+        for f in files:
+            try:
+                with open(f) as file:
+                    domains += file.readlines()
+                logging.debug(f"Ingested domains from file '{file}'")
+            except:
+                logging.debug(f"Could not read file '{file}'")
+        logging.warn(f"Ingested {len(domains)} domains from folder '{filename}'")
+    return [Domain(domain.rstrip()) for domain in domains]
+
+
+def fetch_nested_files(dir):
```

Comment:
why not just use `glob`?

Reply:
happy to use glob if you think its better / more performant